### DISCUSSION TITLE
Retrieving servers info

### DISCUSSION THREAD

**Discussion by JoanGTSQ** [hibiken/asynq#397] [Q&A] [answered]
Hi,
I would like to retrieve a count of all workers running, it's possible?
Thanks

**Comment by hibiken** [comment, marked as answer]
Hi @JoanGTSQ, thank you for posting a question.

There are a few ways:

- via Web UI
- via CLI
- via `Inspector` type (programmatically)

**Note:**
Also, the term *"worker"* is overloaded in Asynq. At macro level, it refers to a process running `asynq.Server`. At micro level, it refers to a goroutine processing a task (e.g. a `Server` can have multiple workers processing multiple tasks). Either way all three methods listed above can show you the count of "workers".

### via Web UI
Navigate to the "Servers" tab in the sidebar:
<img width="1438" alt="Screen Shot 2022-02-16 at 6 02 27 AM" src="https://user-images.githubusercontent.com/10953044/154280357-3fbd8b66-f225-4b7d-b156-91a231c0e7a1.png">

### via CLI
Run the following command:

```sh
asynq server ls
```


### via `Inspector` type (programmatically)
Use [`Inspector.Servers`](https://pkg.go.dev/github.com/hibiken/asynq#Inspector.Servers).

---

Let me know if you have more questions!

**Reply by JoanGTSQ** [reply]
Thanks @hibiken!
I will try it this night and let you if I can have the data I need
